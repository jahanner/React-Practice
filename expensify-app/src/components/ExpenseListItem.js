import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({description, amount, createdAt, props}) => (
  <div>
    <h3>{description}</h3>
    <p>{amount} - {createdAt}</p>
    <button onClick={(e) => {
      props.dispatch(removeExpense(e.target.id));
    }}>Remove</button>
  </div>
);

export default connect()(ExpenseListItem);
