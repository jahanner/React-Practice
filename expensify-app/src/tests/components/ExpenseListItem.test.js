import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses.js";
import ExpenseListItem from "../../components/ExpenseListItem.js";

test("Should render ExpenseListItem from expenses", () => {
  const wrapper = shallow(
    <ExpenseListItem
      description={expenses.description}
      id={expenses.id}
      amount={expenses.amount}
      createdAt={expenses.createdAt}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
