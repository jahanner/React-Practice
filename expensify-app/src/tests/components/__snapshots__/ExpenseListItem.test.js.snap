// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem from expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  >
    <h3 />
  </Link>
  <p>
     - 
  </p>
</div>
`;
